pkg/config/config.go

```diff
@@ -1560,3 +1560,48 @@ func ContainHiddenConfig(s string) bool {
 func GetGlobalKeyspaceName() string {
 	return GetGlobalConfig().KeyspaceName
 }
+
+// OptionDetail is the detail of a config option with its current value and default value.
+type OptionDetail struct {
+	// config name
+	Name string `json:"name"`
+	// current config value
+	Value any `json:"value"`
+	// default config value
+	DefaultValue any `json:"default_value"`
+}
+
+// GetGlobalConfigDetail returns all configuration details.
+func GetGlobalConfigDetail() ([]OptionDetail, error) {
+	cfgMap := make(map[string]interface{})
+	gs, err := GetJSONConfig()
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal([]byte(gs), &cfgMap)
+	if err != nil {
+		return nil, err
+	}
+	cfgItems := FlattenConfigItems(cfgMap)
+
+	j, err := json.Marshal(&defaultConf)
+	if err != nil {
+		return nil, err
+	}
+	dg := make(map[string]interface{})
+	err = json.Unmarshal(j, &dg)
+	if err != nil {
+		return nil, err
+	}
+	defaultCfgItems := FlattenConfigItems(dg)
+
+	result := make([]OptionDetail, 0)
```

Comment:
```suggestion
	result := []OptionDetail{}
```

Reply:
Note that increasing the 0 in the original statement this would not lead to the test failing.